File: 二叉树/bfs 找树左下角的值.go
package 二叉树

//从右向左进行层序遍历，队列中的最后一个点就是树最左下角的点
func findBottomLeftValue(root *TreeNode) int {
	queue := []*TreeNode{root}
	var res int
	for len(queue) != 0 {
		cp := queue[0]
		queue = queue[1:]
		res = cp.Val
		if cp.Right != nil {
			queue = append(queue, cp.Right)
		}
		if cp.Left != nil {
			queue = append(queue, cp.Left)
		}
	}
	return res
}

File: 二叉树/bfs_找树左下角的值.java
package 二叉树;

import java.util.ArrayDeque;

//剑指offer
public class bfs_找树左下角的值 {
    public int findBottomLeftValue(TreeNode root) {
        var queue = new ArrayDeque<TreeNode>();
        queue.add(root);
        int val = 0;
        while (!queue.isEmpty()) {
            var cur = queue.poll();
            if (cur.right != null) queue.add(cur.right);
            if (cur.left != null) queue.add(cur.left);
            val = cur.val;
        }
        return val;
    }
}
